Bearded.Utilities/Monads/Result.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Bearded.Utilities.Monads
+{
+    public readonly struct Result<TResult, TError> : IEquatable<Result<TResult, TError>>
+    {
+        private readonly bool isSuccess;
+        private readonly TResult result;
+        private readonly TError error;
+
+        private Result(bool isSuccess, TResult result, TError error)
+        {
+            this.isSuccess = isSuccess;
+            this.result = result;
+            this.error = error;
+        }
+
+        public TResult ResultOrDefault(TResult @default) => isSuccess ? result : @default;
```

Comment:
It's perhaps slightly funny that we provide these easy ways of ignoring the error entirely. In either case, it got me thinking: should there be an easy way to convert between Result and Maybe?

Reply:
Good idea.

Right now this adds a circular dependency between `Core` and `Monads`, but that will be solved once we move `Result` and `Maybe` to the same namespace.

Reply:
Using extension methods would have prevented the circular dependency, however it would require matching which is less efficient, so I agree with this approach.